cypress/support/components/ScorerModal.js

```diff
@@ -6,45 +6,59 @@
 const helpers = require('../../support/Helpers')
 
 // data-testid="teach-session-admin-train-status" (Running, Completed, Failed)
-export function ClickRefreshScoreButton()       { cy.Get('[data-testid="teach-session-admin-refresh-score-button"]').Click() }
-export function SelectAnAction()                { cy.Get('[data-testid="action-scorer-button-clickable"]').should("be.visible").Click() }
-export function ClickAddActionButton()          { cy.Get('[data-testid="action-scorer-add-action-button"]').Click() }
+export function ClickRefreshScoreButton() { cy.Get('[data-testid="teach-session-admin-refresh-score-button"]').Click() }
+export function SelectAnAction() { cy.Get('[data-testid="action-scorer-button-clickable"]').should("be.visible").Click() }
+export function ClickAddActionButton() { cy.Get('[data-testid="action-scorer-add-action-button"]').Click() }
 
-export function ClickAction(expectedResponse, expectedIndexForActionPlacement)
-{
+export function ClickAction(expectedResponse, expectedIndexForActionPlacement) {
   cy.Get('[data-testid="action-scorer-text-response"]').ExactMatch(expectedResponse)
     .parents('div.ms-DetailsRow-fields').find('[data-testid="action-scorer-button-clickable"]')
     .Click()
   VerifyChatMessage(expectedResponse, expectedIndexForActionPlacement)
 }
 
+export function ClickSessionAction(expectedResponse, expectedIndexForActionPlacement) {
+  cy.Get('[data-testid="action-scorer-session-response"]').ExactMatch(expectedResponse)
+    .parents('div.ms-DetailsRow-fields').find('[data-testid="action-scorer-button-clickable"]')
+    .Click()
+  VerifyAdaptiveChatMessage(expectedResponse, expectedIndexForActionPlacement)
+}
+
 // To verify the last chat utterance leave expectedIndexOfMessage undefined
-export function VerifyChatMessage(expectedMessage, expectedIndexOfMessage)
-{
+export function VerifyChatMessage(expectedMessage, expectedIndexOfMessage) {
   var expectedUtterance = expectedMessage.replace(/'/g, "’")
   cy.Get('[data-testid="web-chat-utterances"]').then(elements => {
-    if(!expectedIndexOfMessage) expectedIndexOfMessage = elements.length - 1
+    if (!expectedIndexOfMessage) expectedIndexOfMessage = elements.length - 1
     cy.wrap(elements[expectedIndexOfMessage]).within(e => {
+//      if (Cypress.$(element).find('div.wc-adaptive-card').length) expectedUtterance = 'EndSession: ' + expectedUtterance
       cy.get('div.format-markdown > p').should('have.text', expectedUtterance)
-    })})
+    })
+  })
+}
+
+export function VerifyAdaptiveChatMessage(expectedMessage, expectedIndexOfMessage) {
+  var expectedUtterance = expectedMessage.replace(/'/g, "’")
+  cy.Get('[data-testid="web-chat-utterances"]').then(elements => {
+    if (!expectedIndexOfMessage) expectedIndexOfMessage = elements.length - 1
+    cy.wrap(elements[expectedIndexOfMessage]).within(e => {
```

Comment:
I'd prefer "element" here instead of "e". I would really like to retain the verbose naming, it is more readable.

Reply:
I don't see how this actually tests that the expected response appears in the chat window, the expectedMessage is never used.

Reply:
Good catch. I opted to keep the test generic know in favor of getting something in the tree as the end session issue was a showstopper. I'll double back in the Verify code once my API and form code tests are ready this week